lib/linkerd2-proxy-api-resolve/src/metadata.rs

```diff
@@ -0,0 +1,76 @@
+use crate::identity;
+use indexmap::IndexMap;
+
+/// Metadata describing an endpoint.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Metadata {
+    /// An endpoint's relative weight.
+    ///
+    /// A weight of 0 means that the endpoint should never be preferred over a
+    /// non 0-weighted endpoint.
+    ///
+    /// The default weight, corresponding to 1.0, is 10,000. This enables us to
+    /// specify weights as small as 0.0001 and as large as 400,000+.
+    ///
+    /// A float is not used so that this type can implement `Eq`.
+    weight: u32,
+
+    /// Arbitrary endpoint labels. Primarily used for telemetry.
+    labels: IndexMap<String, String>,
```

Comment:
This is probably something for another PR, but I wonder if it's worth reassessing our use of `IndexMap`s everywhere. IIRC we observed better performance with `IndexMap` (vs the standard library's `HashMap`) in the past. However, in Rust 1.36.0, the stdlib's `HashMap` was changed to be a reexport of `hashbrown::HashMap` (a SwissTable implementation) which should generally offer better performance. 

We might want to consider switching back to std's `HashMap` in cases where we don't rely on insertion ordering, since performance might be better (and it would maybe speed up the builds very slightly by not pulling another dependency).